### ISSUE TITLE
Reformer pre-training gradient accumulation

### ISSUE THREAD

**Issue by apoorv2904** [lucidrains/reformer-pytorch#97]
Hi, 

I was looking at the reformer pretraining script and noticed that the optimizer.step() is called at every step and not after every gradient_accumulate steps.  Is this a bug or am I missing out something?

**Comment by lucidrains** [comment]
@apoorv2904 that was actually pull-requested in some time ago, early on with the repository was still in flux. feel free to send over a fix!

**Comment by lucidrains** [comment]
@apoorv2904 i put in a quick fix here https://github.com/lucidrains/reformer-pytorch/commit/42a8682ff8e7cec3122eff6febc9087f1c53f370 please reopen if it doesn't work!
